### PR TITLE
Fix: mode-select feature on-off id (CON-998)

### PR DESCRIPTION
**Pull request by hideakitai**
`esp_matter::cluster::mode_select::feature::on_off::get_id()` returns `0x00` now. This should be `0x01`.

### CODE REVIEW

File: components/esp_matter/esp_matter_feature.cpp [hunk 1]
```diff
@@ -3062,7 +3062,7 @@ uint32_t get_id()
 {
     // enum class for DepOnOff is not present in the upstream code.
     // Return the code according to the SPEC
-    return 0x00;
+    return 0x01;
```
Comment (jadhavrohit924) [inline components/esp_matter/esp_matter_feature.cpp:3065]:
```suggestion
    return (uint32_t)ModeSelect::Feature::kOnOff;
```

Comment (hideakitai) [reply components/esp_matter/esp_matter_feature.cpp]: Thanks for your suggestion!

**Review by jadhavrohit924 — APPROVED** [review]
Thanks @hideakitai. We will raise internal MR to merge this

File: components/esp_matter/esp_matter_feature.cpp [hunk 2]
```diff
@@ -3062,7 +3062,7 @@ uint32_t get_id()
 {
     // enum class for DepOnOff is not present in the upstream code.
```
Comment (jadhavrohit924) [inline components/esp_matter/esp_matter_feature.cpp:3063]: Since, OnOff is supported in upstream can you please remove this comment?

Comment (hideakitai) [reply components/esp_matter/esp_matter_feature.cpp]: Thank you, fixed

Comment (jadhavrohit924) [reply components/esp_matter/esp_matter_feature.cpp]: Please remove `// Return the code according to the SPEC` too

File: components/esp_matter/esp_matter_feature.cpp [hunk 3]
```diff
@@ -3060,9 +3060,8 @@ namespace on_off {
 
 uint32_t get_id()
 {
-    // enum class for DepOnOff is not present in the upstream code.
     // Return the code according to the SPEC
```
Comment (jadhavrohit924) [inline components/esp_matter/esp_matter_feature.cpp:3063]: Please remove this line too.

Comment (hideakitai) [reply components/esp_matter/esp_matter_feature.cpp]: ok, removed

**Pull request merged**